dev/build-tool-support/src/main/scala/com/lightbend/lagom/dev/LagomConfig.scala

```diff
@@ -20,7 +20,8 @@ object LagomConfig {
   def actorSystemConfig(name: String) = Map(
     "lagom.akka.dev-mode.actor-system.name" -> s"$name-internal-dev-mode",
     "play.akka.actor-system" -> s"$name-application",
-    "lagom.defaults.cluster.join-self" -> "on"
+    "lagom.defaults.cluster.join-self" -> "on",
+    "akka.discovery.method" -> "lagom-devmode"
```

Comment:
TODO: can pass the class name directly here, so the `reference.conf` changes below can be removed.

Reply:
Reconsidering: looks like we'll need to keep different Java & Scala implementations, to the level of indirection is necessary.